### ISSUE TITLE
TokenMismatchException in compiled.php after Login

### ISSUE THREAD

**Issue by lycotech** [AsgardCms/Platform#95]
Anytime i login it display TokenMismatchException in compiled.php

**Comment by nWidart** [comment]
Hello,

That is because you kept the form open for too long. A TokenMismatchException is expected behaviour in Laravel.

**Comment by Thorazine** [comment]
I have this problem too, but not after a login. Sometimes it just mismatches whilst in the middle of editing a few records without any significant time in between. It almost seems like it's not refreshing the token.
This problem only happens in Asgard for me, never when I use Laravel core.

**Comment by nWidart** [comment]
Asgard doesn't do anything with the tokens, it uses the tokens laravel has set up.

**Comment by Thorazine** [comment]
God your quick in response. Thanks! 

I think there should be some desired action in correspondence to the timeout. Seeing the token times out and a nasty error shows (or te whoops), it would be nice if Asgard catches the error and redirects back (or something) refreshing the token whilst doing it. Worst case somebody loses their input data, but can at least continue in their path. Cause this way my PM shoots in a ticket (which she did) telling me there is an error with a db record (which of course there isn't (sigh)).

**Comment by nWidart** [comment]
I wouldn't not redirect back, since that would just undo the what the token is there for.

That's why I'm saying you can do that yourself, if you really don't care about that protection. You could even completely remove the csrf middleware.

**Comment by Thorazine** [comment]
I've got some more information on this.
My tab with the cms was open all night. When I came back it forced a new login (like it should). I went to edit a record, tried to save, token mismatch. 
This is strange behaviour seeing there were at least 2 pages and a token change (login screen) in between login and form submission.
There is definitely something going on with the token in asgard. I don't think it's a core thing seeing I we have built several projects with our own cms. Never before have we met with the token mismatch this way. We normally only see one with ajax requests. Normal behaviour is a redirect to the login screen seeing token and login timeout together. 
Are you sure you are not refreshing the token somewhere manually or unloading the session which forces a new token. Or something.
